wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -191,6 +201,17 @@ public CacheConnection(final Properties properties) {
     } else {
       this.credentialsProvider = null;
     }
+
+    // Initialize CacheMonitor with telemetry factory
+    if (telemetryFactory != null) {
```

Comment:
Why do you need to do a null check in this layer? 

Can you handle this inside `CacheMonitor.getInstance()`?